src/nvim/mbyte.c

```diff
@@ -1064,17 +1020,25 @@ int utfc_ptr2len_len(const char_u *p, int size)
  */
 int utf_char2len(int c)
 {
-  if (c < 0x80)
```

Comment:
Here it probably best just to keep the existing implementation, possibly returning something else instead of 5,6 (but we need to check callsites what would be best)